autoload/go/fmt.vim

```diff
@@ -52,13 +52,25 @@ endif
 "  this and have VimL experience, please look at the function for
 "  improvements, patches are welcome :)
 function! go#fmt#Format(withGoimport)
-    " save cursor position, folds and many other things
-    let l:curw = {}
-    try
-        mkview!
-    catch
+    if g:go_fmt_experimental == 1
+        " Using winsaveview to save/restore cursor state has the problem of
+        " closing folds on save:
+        "   https://github.com/fatih/vim-go/issues/502
+        " One fix is to use mkview instead. Unfortunately, this sometimes causes
+        " other bad side effects:
+        "   https://github.com/fatih/vim-go/issues/728
+        " and still closes all folds if foldlevel>0:
+        "   https://github.com/fatih/vim-go/issues/732
+        let l:curw = {}
+        try
+            mkview!
+        catch
+            let l:curw=winsaveview()
+        endtry
+    else
```

Comment:
Right, but the original version was like this: https://github.com/fatih/vim-go/pull/691/files So we were saving the cursor position with `winsaveview`. This change is now putting it behind a flag. 

There is already a flag as you know, so we need to put `mkview` behind this: https://github.com/fatih/vim-go/blob/6c1cc914c35735768e54fc4ab0be900a9d62797b/autoload/go/fmt.vim#L64

So final state should be something like:

```
if experimental == 0 
  winsaveview()
else
  wundo...
  mkview...
endif